feature_bit_verifier.go

```diff
@@ -0,0 +1,54 @@
+package taprootassets
+
+import (
+	"context"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/taproot-assets/tapchannel"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// LndFeatureBitVerifier is a struct that verifies that the feature bits of a
+// target connected peer, using our registered lnd node.
+type LndFeatureBitVerifier struct {
+	lnd *lndclient.LndServices
+}
+
+// NewLndFeatureBitVerifier creates a new LndFeatureBitVerifier instance.
+func NewLndFeatureBitVerifier(
+	lnd *lndclient.LndServices) *LndFeatureBitVerifier {
+
+	return &LndFeatureBitVerifier{
+		lnd: lnd,
+	}
+}
+
+// HasFeature returns true if the peer has the given feature bit set. If the
+// peer can't be found, then ErrNoPeer is returned.
+func (l *LndFeatureBitVerifier) HasFeature(ctx context.Context,
+	peerPub btcec.PublicKey, bit lnwire.FeatureBit) (bool, error) {
+
+	peerBytes := route.NewVertex(&peerPub)
+
+	peers, err := l.lnd.Client.ListPeers(ctx)
```

Comment:
Would `GetNodeInfo` also work here? IIUC the info returned from that call should be as up-to-date and what's returned from `ListPeers`, for connected nodes. But AFAICT it doesn't show if we're actually _connected_ to that node.

Reply:
So each time we connect to a peer, we'll get the fresh set of feature bits from the `init` message. Compared to waiting for the next `node_announcement`, this _should_ be more up to date.